sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -894,8 +896,10 @@ object FunctionRegistry {
       since: Option[String] = None): (String, (ExpressionInfo, FunctionBuilder)) = {
     val info = FunctionRegistryBase.expressionInfo[T](name, since)
     val funcBuilder = (expressions: Seq[Expression]) => {
-      assert(expressions.forall(_.resolved), "function arguments must be resolved.")
-      val expr = builder.build(name, expressions)
+      val rearrangedExpressions =
+        SupportsNamedArguments.getRearrangedExpressions[T](expressions, name)
```

Comment:
Does it work? Most `ExpressionBuilder` is an object, e.g. `CeilExpressionBuilder`, `FloorExpressionBuilder`. Does the `SupportsNamedArguments.getRearrangedExpressions` code work when T is an object?

Actually I think we can leave out the support for expression builder in this PR for now. Most expression builder is used only when it has complicated initializer, e.g. build with one param and build with two params. That may require at least two function signatures, or a refactor on the existing code to make it use requires one signature.

Reply:
I see. In that case, I can delete it.